plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -804,14 +854,25 @@ function OPDSBrowser:showDownloads(item)
             end
         end
     end
-    if (#type_buttons % 2 == 1) then -- we need even number of type buttons
-        table.insert(type_buttons, {text = ""})
+    -- handles adding download button(s)
+    if (download_buttons ~= {} and #download_buttons > 0) then
```

Comment:
Ditto.

Hat was the intent of the first part anyway? Can `download_buttons` ever be `nil`?

Reply:
Yes, download_buttons can be nil if there is a file format on the server that can be streamed but not downloaded. For example, this entry in my kavita server can be streamed successfully, but can't be downloaded.
![image](https://user-images.githubusercontent.com/21128449/197844427-1ccba3da-6eb8-474b-910a-65453bd25490.png)


And if you're talking about the `if (#type_buttons % 2 == 1) then` part, that is for putting an empty space if we have an odd number of download buttons greater than 1. So, if there are 3 download buttons it would add the empty space so that there are 2 download buttons per row.

Reply:
Nope, I was indeed talking about the lefthand side of the conditional I quoted ;).

So, just `if download_buttons and #download_buttons > 0`

Reply:
Err, I don't ever see `download_buttons` being `nil`? Empty, yes, but, `nil`, no?

Reply:
Got it, C is actually the language I work with the most (I help teach a class in C) So i guess the habits bleed over :)

Reply:
Then for more context re: the above about comparisons: think of tables as pointers ;).

Reply:
🤦 Why didn't that occur to me?